src/client/dfuse/dfuse_core.c

```diff
@@ -957,6 +957,41 @@ dfuse_cont_open(struct dfuse_projection_info *fs_handle, struct dfuse_pool *dfp,
 	return rc;
 }
 
+/* Set a timer to mark cache entry as valid */
+void
+dfuse_cache_set_time(struct dfuse_inode_entry *ie)
+{
+	struct timespec now;
+
+	clock_gettime(CLOCK_MONOTONIC_COARSE, &now);
+	ie->ie_cache_last_update = now;
+}
+
+bool
+dfuse_cache_get_valid(struct dfuse_inode_entry *ie, double max_age)
+{
+	bool            use = false;
+	struct timespec now;
+	struct timespec left;
+	double          timeout;
+
+	clock_gettime(CLOCK_MONOTONIC_COARSE, &now);
+
+	left.tv_sec  = now.tv_sec - ie->ie_cache_last_update.tv_sec;
+	left.tv_nsec = now.tv_nsec - ie->ie_cache_last_update.tv_nsec;
+	if (left.tv_nsec < 0) {
+		left.tv_sec--;
+		left.tv_nsec += 1000000000;
+	}
+	timeout = max_age - (left.tv_sec + ((double)left.tv_nsec / 1000000000));
```

Comment:
Minor semantic nitpick, not required. The variable name `left` (remaining?) is misleading, since it isn't really the remaining time. "remaining" or "left" only makes sense with respect to `max_age`, which isn't considered until `timeout = ...`. I think `age` or `actual_age` is more self-documenting:
```
age = now - last_update
timeout = max_age - age
```

Reply:
Yeah, I went through several iterations here.  Will update to actual_age or similar.

Reply:
Alternatively https://github.com/daos-stack/daos/pull/9565 builds on this same new code to add caching of metadata but needs to update this function to pass timeout back out so I could do the re-name as part of that work to avoid another spin on this.  That PR is higher priority but is currently causing test-failures hence the desire to move forward on this one in the mean time.

Reply:
Works for me. I don't think a variable name needs a respin :)